### PR TITLE
fix: closing all processes on sigint and sigterm

### PR DESCRIPTION
**Pull request by kostagersh**
Closes CRVD-1158

### CODE REVIEW

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 1]
```diff
@@ -67,4 +68,44 @@ describe("browser sanity", () => {
       await openEditorCliCommand.waitForCommandToEnd();
     })
   );
+
+  test("should show a warning message when closing with unsaved changes and allow to continue working", async () => {
+    process.env.SKIP_UNSAVED_DIALOG = true;
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: do not change it on the global process variable.
instead, pass this to the relevant driver command so it'll execute the child process with this variable

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 2]
```diff
@@ -67,4 +68,44 @@ describe("browser sanity", () => {
...
+
+    const openEditorCliCommand = await cliDriver.openEditor();
+    const editorEditDriver = await connectToLocalEditor(
+      openEditorCliCommand.editorDebugPort
+    );
+    const editDriver = await editorEditDriver.waitForEditor();
+
+    await editDriver.addTextElement();
+
+    let openEditorOutput = "";
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: extract to the command driver ?

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 3]
```diff
@@ -67,4 +68,44 @@ describe("browser sanity", () => {
...
+      cliCloneCommandResult.editorDebugPort
+    );
+
+    const loginDriver = await editorCloneDriver.waitForLogin();
+    await loginDriver.login(getCorvidTestUser());
+
+    await cliCloneCommandResult.waitForCommandToEnd();
+
+    const openEditorCliCommand = await cliDriver.openEditor();
+    const editorEditDriver = await connectToLocalEditor(
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: naming of the different drivers is really confusing

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 4]
```diff
@@ -67,4 +68,44 @@ describe("browser sanity", () => {
...
+    openEditorCliCommand.stdout.on("data", function(data) {
+      openEditorOutput += data.toString();
+    });
+
+    openEditorCliCommand.kill("SIGINT", { forceKillAfterTimeout: false });
+
+    await eventually(() => {
+      expect(openEditorOutput).toContain("You have unsaved changes in editor");
+    });
+    await editDriver.saveLocal();
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: we need to make sure that when this is resolved, the changes were actually saved.
currently, it just clicks the button and resolves.

File: packages/corvid-e2e-tests/test/sanity.e2e.js [hunk 5]
```diff
@@ -67,4 +68,44 @@ describe("browser sanity", () => {
...
+
+    await editorEditDriver.close();
+    await openEditorCliCommand.waitForCommandToEnd();
+    process.env.SKIP_UNSAVED_DIALOG = false;
```
Comment (dor-itzhaki) [inline packages/corvid-e2e-tests/test/sanity.e2e.js]: remove

File: packages/corvid-e2e-tests/test/sanity.e2e.js (hunk 1 above)
Comment (kostagersh) [reply packages/corvid-e2e-tests/test/sanity.e2e.js]: done

File: packages/corvid-e2e-tests/test/sanity.e2e.js (hunk 2 above)
Comment (kostagersh) [reply packages/corvid-e2e-tests/test/sanity.e2e.js]: done

File: packages/corvid-e2e-tests/test/sanity.e2e.js (hunk 5 above)
Comment (kostagersh) [reply packages/corvid-e2e-tests/test/sanity.e2e.js]: done

File: packages/corvid-e2e-tests/test/sanity.e2e.js (hunk 4 above)
Comment (kostagersh) [reply packages/corvid-e2e-tests/test/sanity.e2e.js]: done

File: packages/corvid-e2e-tests/test/sanity.e2e.js (hunk 3 above)
Comment (kostagersh) [reply packages/corvid-e2e-tests/test/sanity.e2e.js]: made some changes

**Pull request merged**